compiler/AST/build.cpp

```diff
@@ -794,7 +794,31 @@ ModuleSymbol* buildModule(const char* name,
   return mod;
 }
 
-
+BlockStmt* buildIncludeModule(const char* name,
+                              bool priv,
+                              bool prototype,
+                              const char* docs) {
+  astlocT loc(chplLineno, yyfilename);
+  ModuleSymbol* mod = parseIncludedSubmodule(name);
+  if (priv && !mod->hasFlag(FLAG_PRIVATE)) {
+    // make the module private (override public)
+    mod->addFlag(FLAG_PRIVATE);
+  }
+  if (mod->hasFlag(FLAG_PRIVATE) && !priv) {
+    USR_FATAL_CONT(loc, "include public module with module declared private");
```

Comment:
This error message doesn't seem very clearly worded for someone who makes this mistake.  What about "cannot make a private module public through an include statement"?  (I'm sure that could be improved upon as well).

While here:  I'd be inclined to make this conditional and the previous an "else" of one another since only one will fire.  Or to use more of a nested conditional a la: `if (priv) { if (!mod->hasFlag(...)) } else { if mod->hasFlag(...) }`  

In any case, a brief "module declared public but included as private: OK" / "module declared private but included as public" comment would probably be helpful.

Reply:
I've fixed that